cmd/pebble/main.go

```diff
@@ -321,11 +321,19 @@ func main() {
 	}()
 
 	if err := run(); err != nil {
+		var status *exitStatus
+		if errors.As(err, &status) {
+			os.Exit(status.code)
+		}
 		fmt.Fprintf(Stderr, errorPrefix+"%v\n", err)
 		os.Exit(1)
 	}
 }
 
+// exitStatus can be returned as an error from commands to have Pebble exit
+// with the given exit code. Pebble's main function also recovers from a
+// panic(&exitStatus{code}), for the rare cases when an error return is not
+// possible.
```

Comment:
This has changed the semantics of what the exitStatus panic actually means, and it's not clear we want that as this is removing features we have now without a tradeoff. It wasn't about it being a rare case, but about avoiding the typical handling of the error result as is observed in the logic above. Note how in the original logic we can have an error message or not, and can have an exit code or not, and these are completely orthogonal to each other. With the new logic, both the panic and the error return do the same thing, and neither of them can have an error message. I suggest returning to the original logic so we have less duplication here and preserve the ability of having both options independently, and then using a panic in the new command just as in cmdRun, as the goal is similar.

Reply:
I'm not sure how this removes features. Note that it's too things: if we want to return an exit code other than 0 or 1 (nil error or non-nil error), or in the rare case where there's no error return, such as the use in `Parser`. I don't know why `Parser` wasn't just made to return an error, but I'll leave that for now.

In any case, I've reverted this change (and updated the comment).

Reply:
Sorry if that wasn't clear above. It removed features because it was impossible for having a custom exitStatus while preserving the error message:

![image](https://user-images.githubusercontent.com/378683/134660340-a483dc1a-98c0-4d34-a5cb-366bd0c755b6.png)

There are definitely other ways of doing that, including the removal of the panic altogether. What felt suboptimal about the change is that it removed the use of the panic, removed the ability to have error message + exit code, and yet kept the handling of the panic around. That set together is what made it seem unwanted.